frontends/web/src/routes/account/summary/accountssummary.tsx

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright 2018 Shift Devices AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, h, RenderableProps } from 'preact';
+import { translate } from 'react-i18next';
+import checkIcon from '../../../assets/icons/check.svg';
+import exportIcon from '../../../assets/icons/download.svg';
+import A from '../../../components/anchor/anchor';
+import { BalanceInterface } from '../../../components/balance/balance';
+import { Header } from '../../../components/layout';
+import * as componentStyle from '../../../components/style.css';
+import { load } from '../../../decorators/load';
+import { TranslateProps } from '../../../decorators/translate';
+import { apiPost } from '../../../utils/request';
+import { AccountInterface } from '../account';
+import { BalancesTable } from './balancestable';
+import { Totals } from './initializeall';
+
+export interface AccountAndBalanceInterface extends AccountInterface {
+    balance: BalanceInterface;
+}
+
+interface AccountSummaryProps {
+    data: Response;
+}
+
+interface State {
+    exported: string;
+}
+
+interface Response {
+    accounts: AccountAndBalanceInterface[];
+    totals: Totals;
+}
+
+type Props = TranslateProps & AccountSummaryProps;
+
+class AccountsSummary extends Component<Props, State> {
+    private groupByCoin(accounts: AccountAndBalanceInterface[], coinCode: string) {
+        return accounts.reduce((accumulator, currentValue) => {
+            const key = currentValue[coinCode];
+            if (!accumulator[key]) {
+                accumulator[key] = [];
+            }
+            accumulator[key].push(currentValue);
+            return accumulator;
+        }, {});
+    }
+
+    public componentDidMount() {
+        this.setState({ exported: ''} );
+    }
+
+    private export = () => {
+        apiPost(`export-account-summary`).then(exported => {
+            this.setState({ exported });
+        });
+    }
+
+    public render(
+        { t, data }: RenderableProps<Props>, { exported }: State,
+    ) {
+            const groupedAccounts = this.groupByCoin(data.accounts, 'coinCode');
```

Comment:
As far as I can tell, `groupByCoin` is used only here. Is there a reason why you parameterized `'coinCode'` with `coinCode: string`?

Reply:
I'll verify once I checked out this PR but I think `groupedAccounts` might be untyped here.